jwst/lib/set_telescope_pointing.py

```diff
@@ -1,231 +1,222 @@
+        use this as the V3 position angle.
+    """
+
+    # Get the SIAF and observation parameters
+    obsstart = float(header['EXPSTART'])
+    obsend = float(header['EXPEND'])
+    siaf = SIAF(
+        v2ref=float(header['V2_REF']),
+        v3ref=float(header['V3_REF']),
+        v3idlyang=float(header['V3I_YANG']),
+        vparity=int(header['VPARITY'])
+    )
+
+    # Get the pointing information
     try:
-        q, j2fgs_matrix, fsmcorr, obstime = get_pointing(obsstart, obsend)
+        pointing = get_pointing(obsstart, obsend)
     except ValueError as exception:
-        ra = pheader['TARG_RA']
-        dec = pheader['TARG_DEC']
-        roll = 0
+        ra = header['TARG_RA']
+        dec = header['TARG_DEC']
+        roll = default_pa_v3
 
         logger.warning(
             'Cannot retrieve telescope pointing.'
             '\n{}'
-            '\nUsing TARG_RA={}, TARG_DEC={} and PA_V3=0 '
-            'to set pointing.'.format(exception, ra, dec)
+            '\nUsing TARG_RA={}, TARG_DEC={} and PA_V3={} '
+            'to set pointing.'.format(exception, ra, dec, roll)
         )
 
-        local_roll = compute_local_roll(roll, ra, dec, v2ref, v3ref)
-        wcsinfo = (ra, dec, local_roll)
-        vinfo = (ra, dec, roll)
-        crval1, crval2, pa_aper_deg = wcsinfo
-        v1_ra_deg, v1_dec_deg, v3_pa_deg = vinfo
-        pa_aper_deg = local_roll - vparity * v3idlyang
+        wcsinfo = WCSRef(ra, dec, roll)
+        vinfo = WCSRef(ra, dec, roll)
+
     else:
         # compute relevant WCS information
-        logger.info('Successful read of engineering quaternions.')
-        logger.debug('q={}'.format(q))
-        logger.debug('j2fgs_matrix={}'.format(j2fgs_matrix))
-        logger.debug('fsmcorr={}'.format(fsmcorr))
-        wcsinfo, vinfo = calc_wcs(v2ref, v3ref, v3idlyang, vparity,
-                                  q, j2fgs_matrix, fsmcorr)
-        crval1, crval2, pa_aper_deg = wcsinfo
-        v1_ra_deg, v1_dec_deg, v3_pa_deg = vinfo
-        local_roll = compute_local_roll(v3_pa_deg, crval1, crval2, v2ref, v3ref)
-
-        logger.info(
-            'Computed coordinates from quaternions:'
-            '\n\tRA = {} DEC={} PA_V3={}'.format(crval1, crval2, v3_pa_deg)
-        )
+        logger.info('Successful read of engineering quaternions:')
+        logger.info('\tPointing = {}'.format(pointing))
+        wcsinfo, vinfo = calc_wcs(pointing, siaf)
+
+    logger.info('Aperture WCS info: {}'.format(wcsinfo))
+    logger.info('V1 WCS info: {}'.format(vinfo))
+
+    # Update V1 pointing
+    header['RA_V1'] = vinfo.ra
+    header['DEC_V1'] = vinfo.dec
+    header['PA_V3'] = vinfo.pa
+
+    # Update Aperture pointing
+    header['CRVAL1'] = wcsinfo.ra
+    header['CRVAL2'] = wcsinfo.dec
+    header['PC1_1'] = -np.cos(wcsinfo.pa * D2R)
+    header['PC1_2'] = np.sin(wcsinfo.pa * D2R)
+    header['PC2_1'] = np.sin(wcsinfo.pa * D2R)
+    header['PC2_2'] = np.cos(wcsinfo.pa * D2R)
```

Comment:
wcsinfo.pa is being computed and then in turn used to compute the PCn_m values, but should we also be storing its value directly in the PA_APER keyword? I don't think anyone (SDP or this script) is populating that keyword yet.